sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala

```diff
@@ -348,4 +364,99 @@ class HiveDDLSuite extends QueryTest with SQLTestUtils with TestHiveSingleton {
       }
     }
   }
+
+  test("create/drop database - location") {
+    val catalog = sqlContext.sessionState.catalog
+    withTempDir { tmpDir =>
+      val dbName = "db1"
+      val tabName = "tab1"
+      val path = "file:" + catalog.createDatabasePath(dbName, Option(tmpDir.toString))
+      val fs = new Path(path).getFileSystem(hiveContext.sessionState.newHadoopConf())
+      withTable(tabName) {
+        assert(tmpDir.listFiles.isEmpty)
+        sql(s"CREATE DATABASE $dbName Location '$tmpDir'")
```

Comment:
The directory will be dropped by the Hive metastore API. The test case verified this behavior. 
https://github.com/gatorsmile/spark/blob/mkdir/sql/hive/src/test/scala/org/apache/spark/sql/hive/execution/HiveDDLSuite.scala#L394